### PR TITLE
Function Test rebasing

### PR DESCRIPTION
**Pull request by BigUstad**
Removing RemoveObject/RemoveObjects calls as TearDown takes care of it.

Signed-off-by: Priyank Raj <priyank.raj@gatech.edu>

### CODE REVIEW

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 1]
```diff
@@ -695,12 +695,14 @@ internal async static Task PutObject_Test1(MinioClient minio)
             {
                 await Setup_Test(minio, bucketName);
                 await PutObject_Tester(minio, bucketName, objectName, null, contentType, 0, null, rsg.GenerateStreamFromSeed(1 * KB));
-                await TearDown(minio, bucketName);
                 new MintLogger(nameof(PutObject_Test1), putObjectSignature, "Tests whether PutObject passes for small object", TestStatus.PASS, (DateTime.Now - startTime), args:args).Log();
             }
             catch (Exception ex)
             {
                 new MintLogger(nameof(PutObject_Test1), putObjectSignature, "Tests whether PutObject passes for small object", TestStatus.FAIL, (DateTime.Now - startTime), "", ex.Message, ex.ToString(), args:args).Log();
+            }
```
Comment (kannappanr) [inline Minio.Functional.Tests/FunctionalTest.cs:703]: catch needs to throw

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 2]
```diff
@@ -722,12 +724,14 @@ internal async static Task PutObject_Test2(MinioClient minio)
             {
                 await Setup_Test(minio, bucketName);
                 await PutObject_Tester(minio, bucketName, objectName, null, contentType, 0, null, rsg.GenerateStreamFromSeed(6 * MB));
-                await TearDown(minio, bucketName);
                 new MintLogger(nameof(PutObject_Test2), putObjectSignature, "Tests whether multipart PutObject passes", TestStatus.PASS, (DateTime.Now - startTime), args:args).Log();
             }
             catch (Exception ex)
             {
                 new MintLogger(nameof(PutObject_Test2), putObjectSignature, "Tests whether multipart PutObject passes", TestStatus.FAIL, (DateTime.Now - startTime), "", ex.Message, ex.ToString(), args:args).Log();
+            }
```
Comment (kannappanr) [inline Minio.Functional.Tests/FunctionalTest.cs:732]: catch needs to throw

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 3]
```diff
@@ -750,12 +754,14 @@ internal async static Task PutObject_Test3(MinioClient minio)
             {
                 await Setup_Test(minio, bucketName);
                 await PutObject_Tester(minio, bucketName, objectName, null, contentType, 0, null, rsg.GenerateStreamFromSeed(1 * KB));
-                await TearDown(minio, bucketName);
                 new MintLogger(nameof(PutObject_Test3), putObjectSignature, "Tests whether PutObject with custom content-type passes", TestStatus.PASS, (DateTime.Now - startTime), args:args).Log();
             }
             catch (Exception ex)
             {
                 new MintLogger(nameof(PutObject_Test3), putObjectSignature, "Tests whether PutObject with custom content-type passes", TestStatus.FAIL, (DateTime.Now - startTime), "", ex.Message, ex.ToString(), args:args).Log();
+            }
```
Comment (kannappanr) [inline Minio.Functional.Tests/FunctionalTest.cs:762]: catch needs to throw

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 4]
```diff
@@ -822,14 +828,16 @@ internal async static Task PutObject_Test5(MinioClient minio)
...
                 await Setup_Test(minio, bucketName);
                 await PutObject_Tester(minio, bucketName, objectName, null, null, 0, null, rsg.GenerateStreamFromSeed(1));
-                await TearDown(minio, bucketName);
                 new MintLogger(nameof(PutObject_Test5), putObjectSignature, "Tests whether PutObject with no content-type passes for small object", TestStatus.PASS, (DateTime.Now - startTime), args:args).Log();
             }
             catch (Exception ex)
             {
-                await TearDown(minio, bucketName);
                 new MintLogger(nameof(PutObject_Test5), putObjectSignature, "Tests whether PutObject with no content-type passes for small object", TestStatus.FAIL, (DateTime.Now - startTime), "", ex.Message, ex.ToString(), args:args).Log();
             }
```
Comment (kannappanr) [inline Minio.Functional.Tests/FunctionalTest.cs:836]: missing throw

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 5]
```diff
@@ -867,20 +875,17 @@ internal async static Task PutObject_Test7(MinioClient minio)
...
             {
-                RemoveObjectArgs rmArgs = new RemoveObjectArgs()
-                                                    .WithBucket(bucketName)
-                                                    .WithObject(objectName);
-
-                await minio.RemoveObjectAsync(rmArgs);
-                await TearDown(minio, bucketName);
                 new MintLogger(nameof(PutObject_Test7), putObjectSignature, "Tests whether PutObject with unknown stream-size passes", TestStatus.FAIL, (DateTime.Now - startTime), "", ex.Message, ex.ToString(), args:args).Log();
             }
+            finally
```
Comment (kannappanr) [inline Minio.Functional.Tests/FunctionalTest.cs:885]: missing throw

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 6]
```diff
@@ -917,20 +922,17 @@ internal async static Task PutObject_Test8(MinioClient minio)
...
             {
-                RemoveObjectArgs rmArgs = new RemoveObjectArgs()
-                                                    .WithBucket(bucketName)
-                                                    .WithObject(objectName);
-
-                await minio.RemoveObjectAsync(rmArgs);
-                await TearDown(minio, bucketName);
                 new MintLogger(nameof(PutObject_Test8), putObjectSignature, "Tests PutObject where unknown stream sends 0 bytes", TestStatus.FAIL, (DateTime.Now - startTime), "", ex.Message, ex.ToString(), args:args).Log();
             }
+            finally
```
Comment (kannappanr) [inline Minio.Functional.Tests/FunctionalTest.cs:932]: catch is missing throw

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 7]
```diff
@@ -1153,13 +1150,12 @@ internal async static Task PutGetStatEncryptedObject_Test3(MinioClient minio)
             }
             catch (Exception ex)
             {
-                RemoveObjectArgs rmArgs = new RemoveObjectArgs()
-                                                    .WithBucket(bucketName)
-                                                    .WithObject(objectName);
-                await minio.RemoveObjectAsync(rmArgs);
-                await TearDown(minio, bucketName);
                 new MintLogger("PutGetStatEncryptedObject_Test3", putObjectSignature, "Tests whether Put/Get/Stat multipart upload with encryption passes", TestStatus.FAIL, (DateTime.Now - startTime), "", ex.Message, ex.ToString(), args).Log();
             }
```
Comment (kannappanr) [inline Minio.Functional.Tests/FunctionalTest.cs:1154]: catch is missing throw

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 8]
```diff
@@ -1393,36 +1389,38 @@ internal async static Task CopyObject_Test2(MinioClient minio)
...
+                                                            .WithBucket(destBucketName)
+                                                            .WithObject(destObjectName);
 
+                await minio.CopyObjectAsync(copyObjectArgs);
+            }
+            catch (MinioException ex)
+            {
+                Assert.IsTrue(ex.Message.Contains("MinIO API responded with message=At least one of the pre-conditions you specified did not hold"));
                 new MintLogger("CopyObject_Test2", copyObjectSignature, "Tests whether CopyObject with Etag mismatch passes", TestStatus.PASS, (DateTime.Now - startTime), args:args).Log();
             }
```
Comment (kannappanr) [inline Minio.Functional.Tests/FunctionalTest.cs:1420]: missing throw in catch

Comment (kannappanr) [reply Minio.Functional.Tests/FunctionalTest.cs:1420]: is the Assert code needed here?

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 9]
```diff
@@ -2329,21 +2327,27 @@ internal async static Task GetObject_Test2(MinioClient minio)
...
+            try
+            {
+                GetObjectArgs getObjectArgs = new GetObjectArgs()
+                                                        .WithBucket(bucketName)
+                                                        .WithObject(objectName)
+                                                        .WithFile(fileName);
+                await minio.GetObjectAsync(getObjectArgs);
                 new MintLogger("GetObject_Test2", getObjectSignature, "Tests for non-existent GetObject", TestStatus.PASS, (DateTime.Now - startTime), args:args).Log();
             }
+            catch (ObjectNotFoundException ex)
```
Comment (kannappanr) [inline Minio.Functional.Tests/FunctionalTest.cs:2347]: what does this exception do?

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 10]
```diff
@@ -3109,50 +3113,39 @@ internal async static Task PresignedGetObject_Test2(MinioClient minio)
...
+                ObjectStat stats = await minio.StatObjectAsync(statObjectArgs);
+                PresignedGetObjectArgs preArgs = new PresignedGetObjectArgs()
+                                                                .WithBucket(bucketName)
+                                                                .WithObject(objectName)
+                                                                .WithExpiry(0);
+                string presigned_url = await minio.PresignedGetObjectAsync(preArgs);
+            }
+            catch (InvalidExpiryRangeException)
+            {
+                new MintLogger("PresignedGetObject_Test2", presignedGetObjectSignature, "Tests whether PresignedGetObject url retrieves object from bucket when invalid expiry is set.", TestStatus.PASS, (DateTime.Now - startTime), args:args).Log();
```
Comment (kannappanr) [inline Minio.Functional.Tests/FunctionalTest.cs:3139]: missing throw

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 11]
```diff
@@ -3109,50 +3113,39 @@ internal async static Task PresignedGetObject_Test2(MinioClient minio)
...
             catch (Exception ex)
             {
-                RemoveObjectArgs rmArgs = new RemoveObjectArgs()
-                                                    .WithBucket(bucketName)
-                                                    .WithObject(objectName);
-
-                await minio.RemoveObjectAsync(rmArgs);
-                await TearDown(minio, bucketName);
                 new MintLogger("PresignedGetObject_Test2", presignedGetObjectSignature, "Tests whether PresignedGetObject url retrieves object from bucket when invalid expiry is set.", TestStatus.FAIL, (DateTime.Now - startTime), ex.Message, ex.ToString(), args:args).Log();
             }
```
Comment (kannappanr) [inline Minio.Functional.Tests/FunctionalTest.cs:3144]: missing throw

File: Minio.Functional.Tests/FunctionalTest.cs (hunk 8 above)
Comment (BigUstad) [reply Minio.Functional.Tests/FunctionalTest.cs:1429]: > is the Assert code needed here?

Yes. Checks for pre-condition, specifically.

File: Minio.Functional.Tests/FunctionalTest.cs (hunk 9 above)
Comment (BigUstad) [reply Minio.Functional.Tests/FunctionalTest.cs:2356]: Checks `GetObjectAsync` for an object that doesn't exist, verifies that the exception is indeed thrown.

File: Minio.Functional.Tests/FunctionalTest.cs (hunk 10 above)
Comment (BigUstad) [reply Minio.Functional.Tests/FunctionalTest.cs:3149]: `TestStatus.PASS`

File: Minio.Functional.Tests/FunctionalTest.cs (hunk 8 above)
Comment (kannappanr) [reply Minio.Functional.Tests/FunctionalTest.cs:1429]: I am not sure if this is correct. 
```
Assert.IsTrue(ex.Message.Contains("MinIO API responded with message=At least one of the pre-conditions you specified did not hold"))
```

File: Minio.Functional.Tests/FunctionalTest.cs (hunk 9 above)
Comment (kannappanr) [reply Minio.Functional.Tests/FunctionalTest.cs:2356]: if that is the case, then `Asset.AreEqual` is not needed, you can simply log `PASS` in the log. The logger statement after `GetObjectAsync` should not be present. You should simply throw an exception which will be caught in the exception handler.

File: Minio.Functional.Tests/FunctionalTest.cs (hunk 10 above)
Comment (kannappanr) [reply Minio.Functional.Tests/FunctionalTest.cs:3149]: the throw should be present after the line
```
string presigned_url = await minio.PresignedGetObjectAsync(preArgs);
```
as it is not expected to succeed in this case

File: Minio.Functional.Tests/FunctionalTest.cs (hunk 9 above)
Comment (BigUstad) [reply Minio.Functional.Tests/FunctionalTest.cs:2356]: Changed.

File: Minio.Functional.Tests/FunctionalTest.cs (hunk 10 above)
Comment (BigUstad) [reply Minio.Functional.Tests/FunctionalTest.cs:3149]: Changed.

File: Minio.Functional.Tests/FunctionalTest.cs (hunk 8 above)
Comment (BigUstad) [reply Minio.Functional.Tests/FunctionalTest.cs:1429]: OK, changed.

**Review by kannappanr — APPROVED** [review]
LGTM

**Pull request merged**